helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -249,28 +336,30 @@ private String deploySolo(final String heliosHost) throws HeliosDeploymentExcept
     //TODO(negz): Don't make this.env immutable so early?
     final List<String> env = new ArrayList<>();
     env.addAll(this.env);
-    env.add("HELIOS_NAME=" + HELIOS_NAME_PREFIX + this.namespace);
+    env.add("HELIOS_NAME=" + HELIOS_NAME_PREFIX + namespace);
     env.add("HOST_ADDRESS=" + heliosHost);
 
     final String heliosPort = String.format("%d/tcp", HELIOS_MASTER_PORT);
+    final String watchdogPort = String.format("%d/tcp", HELIOS_SOLO_WATCHDOG_PORT);
     final Map<String, List<PortBinding>> portBindings = ImmutableMap.of(
-            heliosPort, singletonList(PortBinding.of("0.0.0.0", "")));
+        heliosPort, singletonList(PortBinding.randomPort("0.0.0.0")),
+        watchdogPort, singletonList(PortBinding.randomPort("0.0.0.0")));
     final HostConfig hostConfig = HostConfig.builder()
-            .portBindings(portBindings)
-            .binds(binds)
-            .build();
+        .portBindings(portBindings)
+        .binds(binds)
+        .build();
     final ContainerConfig containerConfig = ContainerConfig.builder()
-            .env(ImmutableList.copyOf(env))
-            .hostConfig(hostConfig)
-            .image(HELIOS_IMAGE)
-            .build();
+        .env(ImmutableList.copyOf(env))
+        .hostConfig(hostConfig)
+        .image(HELIOS_IMAGE)
+        .build();
 
     log.info("starting container for helios-solo with image={}", HELIOS_IMAGE);
 
     final ContainerCreation creation;
     try {
-      dockerClient.pull(HELIOS_IMAGE);
-      final String containerName = HELIOS_CONTAINER_PREFIX + this.namespace;
+      //dockerClient.pull(HELIOS_IMAGE);
```

Comment:
It needs to be commented out for now because it'll pull down the spotify/helios-solo:latest image from docker hub which doesn't have the watchdog. That's also the reason I had to modify circle.sh to build and tag a spotify/helios-solo:latest image. We can put these back later I think.